plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java

```diff
@@ -510,6 +515,30 @@ public boolean isDropTableDangerous() {
             ((Drop) statement).getName() != null && ((Drop) statement).getType().equalsIgnoreCase("table");
     }
 
+    public boolean isModifyingSelect() {
+        if (getType() == SQLQueryType.UNKNOWN) {
+            return false;
+        }
+        if (statement instanceof Select) {
+            SelectBody selectBody = ((Select) statement).getSelectBody();
+            if (selectBody instanceof PlainSelect plainSelectBody) {
+                return plainSelectBody.isForUpdate() || plainSelectBody.getIntoTables() != null;
+            }
+            return false;
+        } else {
+            return true;

```

Comment:
Just to check - is it expected to have "true" on non-select statement in isModifyingSelect method?

Reply:
I think, I misunderstood this method.
P.S. I haven't changed it, just moved and renamed.